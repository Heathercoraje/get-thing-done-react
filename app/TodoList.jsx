import React from "react";

const TodoList = ({ todos, deleteTodo }) => {
  const completeTodo = event => {
    document.getElementById(event.target.value).classList.toggle("strike");
  };
  const Renderlist = todos => {
    if (todos.length === 1) {
      return <p>Start adding task </p>;
    }
    return todos.map(todo => {
      if (todo.id === 0) {
        return "";
      }
      return (
        <div key={todo.id}>
          <li key={todo.id} id={todo.id}>
            {todo.todo}
            <button onClick={deleteTodo} value={todo.id}>
              Delete
            </button>
            <button onClick={completeTodo} value={todo.id}>
              Complete
            </button>
          </li>
        </div>
      );
    });
  };
  return (
    <div className="rightList">
      <h1> Today </h1>
      <p> List of things to do today</p>
      <div>{Renderlist(todos)}</div>
    </div>
  );
};

export default TodoList;
