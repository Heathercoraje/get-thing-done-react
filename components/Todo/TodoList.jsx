import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Desc } from './Desc';

const Wrapper = styled.div`
	width: 100%;
	overflow-y: auto;
	overflow-x: hidden;
`;

const TodoList = props => {
	const todos = props.todos;
	const selected = props.selected;
	const name = props.name;
	const descText = props.categories[selected].description;
	const Renderlist = !todos
		? ''
		: todos.map(todo => (
				<TodoItem
					todo={todo.todo}
					key={todo.id}
					id={todo.id}
					isDone={todo.isDone}
					value={todo.id}
					delete={props.delete}
					complete={props.complete}
				/>
			));
	return (
		<Wrapper>
			<Desc name={name} description={descText} HandleDesc={props.HandleDesc} />
			<div>{Renderlist}</div>
		</Wrapper>
	);
};

TodoList.propTypes = {
	todos: PropTypes.array,
	categories: PropTypes.array,
	selected: PropTypes.number,
	name: PropTypes.string,
	delete: PropTypes.func,
	complete: PropTypes.func,
	HandleDesc: PropTypes.func
}
const TodoItem = props => (
	<div key={props.id} className="list-todo">
		<button className="checkbox" onClick={props.delete} value={props.value}>
			&#9744;
		</button>
		<li
			id={props.id}
			onClick={props.complete}
			value={props.value}
			style={!props.isDone ?
			null : { textDecoration: 'line-through' }}
		>
			{props.todo}
		</li>
	</div>
);

TodoItem.propTypes = {
	todo: PropTypes.string,
	id: PropTypes.number,
	value: PropTypes.string,
	isDone: PropTypes.bool,
	delete: PropTypes.func,
	complete: PropTypes.func
}

export { TodoList, TodoItem };
