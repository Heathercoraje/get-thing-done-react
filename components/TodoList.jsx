import React, { Component } from 'react';
import styled from 'styled-components';
import Desc from './Desc';

const Wrapper = styled.div`
	width: 100%;
`;

class TodoList extends Component {
  constructor(props) {
    super(props);
    this.state = {
      editMode: false
    };
  }
  render() {
    const todos = this.props.todos;
    // console.log(this.props.todos.length, this.props.categories.length);
    const Renderlist = (todos) => {
      if (!todos) {
        return '';
      }
      return todos.map(todo => (
        <div key={todo.id} className="list-todo">
          <button className="checkbox" onClick={this.props.delete} value={todo.id}>
						&#9744;
          </button>
          <li
            id={todo.id}
            onClick={this.props.complete}
            value={todo.id}
            style={!todo.isDone ? null : { textDecoration: 'line-through' }}
          >
            {todo.todo}
          </li>
        </div>
      ));
    };

    return (
      <Wrapper>
        <Desc
          name={this.props.categories[this.props.selected].category}
          desc={
            this.props.categories[this.props.selected].desc === ''
              ? 'Add description of category'
              : this.props.categories[this.props.selected].desc
          }
          HandleDesc={this.props.HandleDesc}
        />
        <div>{Renderlist(todos)}</div>
      </Wrapper>
    );
  }
}

export default TodoList;
