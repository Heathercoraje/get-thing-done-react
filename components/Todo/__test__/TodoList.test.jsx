import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { TodoList, TodoItem } from '../TodoList';

test('TodoList should render empty Renderlist when given empty todo', () => {
	const categories = [{ id: 0, name: 'todo', description: 'things to do' }];
	const todos = [];
	const name = 'todo';
	const selected = 0;
	const component = shallow(
		<TodoList
      categories={categories}
			todos={todos}
			name={name}
			selected={selected}
		/>
	);
	const tree = toJson(component);
	expect(component.find('TodoItem').length).toBe(0);
	expect(tree).toMatchSnapshot();
});

test('TodoList should render one TodoItem', () => {
	const categories = [{ id: 0, name: 'todo', description: 'things to do' }];
	const todos = [{ id: 0, todo: 'be awesome', isDone: false }];
	const name = 'todo';
	const selected = 0;
	const component = shallow(
		<TodoList
			categories={categories}
			todos={todos}
			name={name}
			selected={selected}
		/>
	);
	const tree = toJson(component);
	expect(tree).toMatchSnapshot();
	expect(component.find('TodoItem').length).toBe(1);
});
