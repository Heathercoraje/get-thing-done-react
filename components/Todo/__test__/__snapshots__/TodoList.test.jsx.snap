// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList should render empty Renderlist when given empty todo 1`] = `
<styled.div>
  <Desc
    description="things to do"
    name="todo"
  />
  <div />
</styled.div>
`;

exports[`TodoList should render one TodoItem 1`] = `
<styled.div>
  <Desc
    description="things to do"
    name="todo"
  />
  <div>
    <TodoItem
      id={0}
      isDone={false}
      key="0"
      todo="be awesome"
      value={0}
    />
  </div>
</styled.div>
`;
